File: Java/ReversedLinkedList.java
import  java.util.Scanner;

class Node{
    int data;
    Node next;

    Node(int data){
        this.data = data;
        this.next = null;
    }
}

class LinkedList{
    Node head = null;

    void inslast(int data){
       Node newNode = new Node(data);

       if(head == null){
        head = newNode;
        return;
       }

       Node temp = head;

       while(temp.next != null){
        temp = temp.next;
       }
       temp.next = newNode;
    }

    void display(){
        Node temp = head;

        while(temp != null){
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.print("Null");
        System.out.println();
    }

    void revlinkedlist(int data){

        Node temp = head;

        Node Prev = null;
        Node Current = head;
        Node Next = null;

        while(Current != null){
            Next = Current.next;
            Current.next = Prev;
            Prev  = Current;
            Current = Next;
        }

        head = Prev;

    }
}

class ReversedLinkedList{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        LinkedList list = new LinkedList();

        for(int i=0; i<n; i++ ){
            int data = sc.nextInt();
            list.inslast(data);
        }

        list.display(); 

          for(int i=0; i<n; i++ ){
            int data = sc.nextInt();
            list.revlinkedlist(data);
        }

        list.display();
    }
}

File: Python/LinkedList/ReversedLinkedList.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def insert_last(self,data):
        new_node = Node(data)

        if self.head is None:
            self.head = new_node
            return

        temp = self.head

        while temp.next:
            temp = temp.next

        temp.next = new_node

    def reverse(self):
        prev = None
        current = self.head

        while current is not None:
            nexxt = current.next
            current.next = prev
            prev = current 
            current = nexxt

        self.head =  prev


    def display(self):
        temp = self.head

        while temp:
            print(temp.data, end=" -> ")
            temp = temp.next
        print("Null")

ll = LinkedList()


a = list(map(int,input().split()))

for num in a:
    ll.insert_last(num)

ll.display()  

ll.reverse()

ll.display()